### PR TITLE
pkgconfig: Add include directories from internal deps in -unintalled.pc

### PR DESCRIPTION
**Pull request by xclaesse**
Fixes: #8651

### CODE REVIEW

**Comment by dnicolodi** [comment]
Typo in the commit message: `-unintalled.pc` should be `-uninstalled.pc` (missing `s`).

**Comment by xclaesse** [comment]
> Typo in the commit message: `-unintalled.pc` should be `-uninstalled.pc` (missing `s`).

thanks, fixed.

File: test cases/common/44 pkgconfig-gen/test.json [hunk 1]
```diff
@@ -12,9 +14,12 @@
     {"type": "file", "file": "usr/lib/pkgconfig/simple3.pc"},
     {"type": "file", "file": "usr/lib/pkgconfig/simple5.pc"},
     {"type": "file", "file": "usr/lib/pkgconfig/simple6.pc"},
+    {"type": "file", "file": "usr/lib/pkgconfig/simple7.pc"},
     {"type": "file", "file": "usr/lib/pkgconfig/ct.pc"},
     {"type": "file", "file": "usr/lib/pkgconfig/ct0.pc"},
     {"type": "file", "file": "usr/share/pkgconfig/libhello_nolib.pc"}
+
```
Comment (tristan957) [inline test cases/common/44 pkgconfig-gen/test.json:21]: This looks unnecessary

**Review by tristan957 — COMMENTED** [review]
What is missing for this?

File: test cases/common/44 pkgconfig-gen/test.json (hunk 1 above)
Comment (xclaesse) [reply test cases/common/44 pkgconfig-gen/test.json]: Fixed

**Comment by xclaesse** [comment]
> What is missing for this?

Rebased and it's just missing a review :)

File: mesonbuild/modules/pkgconfig.py [hunk 2]
```diff
@@ -99,14 +99,15 @@ class DependenciesHelper:
     def __init__(self, state: ModuleState, name: str, metadata: T.Dict[str, MetaData]) -> None:
         self.state = state
         self.name = name
+        self.metadata = metadata
```
Comment (tristan957) [inline mesonbuild/modules/pkgconfig.py:102]: Any reason this was moved?

**Review by tristan957 — COMMENTED** [review]
Nothing seems obviously wrong to me. Mind posting a before and after of a `-uninstalled.pc` file?

**Review by dcbaker — APPROVED** [review]

**Pull request merged**